graylog2-web-interface/src/views/logic/views/ViewTransformer.js

```diff
@@ -0,0 +1,32 @@
+// @flow strict
+import { List, Map } from 'immutable';
+import Widget from 'views/logic/widgets/Widget';
+
+import type { ViewStateMap } from './View';
+import View from './View';
+import ViewState from './ViewState';
+
+const ViewTransformer = (searchView: View): View => {
+  const newViewStateMap: ViewStateMap = (searchView.state || Map.of()).map((viewState: ViewState, queryId: string) => {
+    const { timerange, query, filter = List.of() } = searchView.search.queries.find(q => q.id === queryId);
+
+    const widgets: List<Widget> = viewState.widgets.map((widget: Widget) => {
+      return widget.toBuilder()
+        .timerange(timerange)
+        .query(query)
+        .streams(filter.filter(value => value instanceof Map && value.get('type') === 'stream')
```

Comment:
`instanceof` will not work across plugin boundaries. I would check if `value` is defined and assume it is an immutable map if it is. If you want to play real safe, you can also do a `new Map(value).get('type') === 'stream'`. 

P.S.: We should probably add an ESLint rule that forbids using `instanceof`.